src/SIL.LCModel.Core/SpellChecking/SpellEngine.cs

```diff
@@ -107,13 +140,23 @@ public bool IsVernacular
 		/// <returns></returns>
 		public ICollection<string> Suggest(string badWord)
 		{
+			#if __MonoCS__
 			IntPtr pointerToAddressStringArray;
 			int resultCount = Hunspell_suggest(m_hunspellHandle, MarshallAsUtf8Bytes(badWord), out pointerToAddressStringArray);
 			if (pointerToAddressStringArray == IntPtr.Zero)
 				return new string[0];
 			var results = MarshalUnmananagedStrArray2ManagedStrArray(pointerToAddressStringArray, resultCount);
 			Hunspell_free_list(m_hunspellHandle, ref pointerToAddressStringArray, resultCount);
 			return results;
+			#else
+			List<string> suggestions = m_hunspellHandle.Suggest(MarshallAsUtf8Bytes(badWord));
+			int resultCount = suggestions.Count;
+			if (resultCount == 0)
+				return new string[0];
+
+			var results = suggestions;
+			return results;
```

Comment:
this entire section could be replaced with `return m_hunspellHandle.Suggest(MarshallAsUtf8Bytes(badWord))`